envoy/upstream/load_balancer.h

```diff
@@ -151,8 +154,9 @@ class LoadBalancerContext {
 
   /* Called by the load balancer when asynchronous host selection completes
    * @param host supplies the upstream host selected
+   * @param details gives optional details about the resolution success/failure.
    */
-  virtual void onAsyncHostSelection(HostConstSharedPtr&& host) PURE;
+  virtual void onAsyncHostSelection(HostConstSharedPtr&& host, std::string details) PURE;
```

Comment:
Should this perhaps be `std::string&&` to avoid a copy?

Reply:
SGTM